### PR TITLE
FIX JENKINS-38248 Log should only be linkable from its line number

### PR DESCRIPTION
**Pull request by scherler**
# Description

See [JENKINS-38248](https://issues.jenkins-ci.org/browse/JENKINS-38248).

https://youtu.be/lCQ6SyFZGZg

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

File: blueocean-dashboard/src/main/js/components/Step.jsx [hunk 1]
```diff
@@ -160,13 +168,14 @@ export default class Node extends Component {
...
-const { object, func, string, bool } = PropTypes;
+const { object, func, string, bool, shape } = PropTypes;
 Node.propTypes = {
     node: object.isRequired,
     followAlong: bool,
     logs: object,
     location: object,
     fetchLog: func,
     nodesBaseUrl: string,
+    router: shape,
```
Comment (michaelneale) [inline blueocean-dashboard/src/main/js/components/Step.jsx]: "router" is shape?

**Comment by michaelneale** [comment]
Looks pretty good to me - seems to work as advertised. From what I can tell from the ATH it is failed (not sure how to correlate a CI run with a PR exactly though - cc @tfennelly). 

I do see the odd error when opening a MB pipeline from a favourite: 

```
http://localhost:8080/jenkins/blue/rest/organizations/jenkins/pipelines/keith/runs/15/ Failed to load resource: the server responded with a status of 404 (Not Found)
```
which seems related to something else, not likely this (doesn't affect rendering). 

So :bee: for functionality but needs a code review.

File: blueocean-dashboard/src/main/js/components/Step.jsx (hunk 1 above)
Comment (scherler) [inline blueocean-dashboard/src/main/js/components/Step.jsx]: Yeah, monday I updated the eslint rules and in the new ones you cannot use "object, array and any" as proptypes anymore, so I guess that was still in the back of my head

Comment (michaelneale) [inline blueocean-dashboard/src/main/js/components/Step.jsx]: just curious what "shape" means

File: blueocean-dashboard/src/main/js/components/LogConsole.jsx [hunk 2]
```diff
@@ -142,29 +141,42 @@ export class LogConsole extends Component {
...
                 </div>}
-                { lines.map((line, index) => <p key={index + 1} id={`${prefix}log-${index + 1}`}>
-                    <a
-                      key={index + 1}
-                      href={`#${prefix || ''}log-${index + 1}`}
-                      name={`${prefix}log-${index + 1}`}
-                    >{line}
-                    </a>
-                </p>)}</code>
+                <table className="highlight">
```
Comment (michaelneale) [inline blueocean-dashboard/src/main/js/components/LogConsole.jsx]: (in a good way)

File: blueocean-dashboard/src/main/js/components/Step.jsx (hunk 1 above)
Comment (scherler) [inline blueocean-dashboard/src/main/js/components/Step.jsx]: https://facebook.github.io/react/docs/reusable-components.html

```  
// An object taking on a particular shape
  optionalObjectWithShape: React.PropTypes.shape({
    color: React.PropTypes.string,
    fontSize: React.PropTypes.number
  }),
```

**Comment by scherler** [comment]
@michaelneale from the look on the error that is not related to our change.

```
 creating pipeline job stagesClassic stages-with-wait.groovy
2016-10-06 00:59:51,660 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -  [0;31m???[0m Timed out while waiting for element <body.page-head-removed> to be present for 20000 milliseconds. [1;37m - expected [0;32m"found"[0m[0m but got: [0;31m"not found"[0m
2016-10-06 00:59:51,660 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [0;90m    at Timeout._onTimeout (/home/ubuntu/slave/workspace/ATH-Jenkinsfile/master/src/main/js/custom_commands/removePageHead.js:50:18)
2016-10-06 00:59:51,660 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -     at tryOnTimeout (timers.js:232:11)
2016-10-06 00:59:51,660 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -     at Timer.listOnTimeout (timers.js:202:5)[0m
2016-10-06 00:59:51,660 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-10-06 00:59:51,731 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-10-06 00:59:51,731 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [0;31mFAILED: [0m [0;31m1[0m assertions failed 
```

**Comment by michaelneale** [comment]
@scherler well if you can convince the ATH to run (it has run successfully on master recently) then go ahead with this change, LGTM

**Comment by scherler** [comment]
@michaelneale I am testing local and see the same, I am investigating more

**Comment by scherler** [comment]
Yeah, good catch @michaelneale 

```
        var aXpath = '//code/p/a[1]';
```

that wont work anymore

**Pull request merged**